pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -122,25 +147,31 @@ func (s *Service) Handshake(stream p2p.Stream, peerMultiaddr ma.Multiaddr, peerI
 		return nil, err
 	}
 
-	if err := w.WriteMsgWithTimeout(messageTimeout, &pb.Ack{
+	if err := w.WriteMsgWithTimeout(MessageTimeout, &pb.Ack{
 		Address: &pb.BzzAddress{
 			Underlay:  advertisableUnderlayBytes,
 			Overlay:   bzzAddress.Overlay.Bytes(),
 			Signature: bzzAddress.Signature,
 		},
-		NetworkID: s.networkID,
-		Light:     s.lightNode,
+		NetworkID:      s.networkID,
+		Light:          s.lightNode,
+		WelcomeMessage: s.welcomeMessage,
 	}); err != nil {
 		return nil, fmt.Errorf("write ack message: %w", err)
 	}
 
 	s.logger.Tracef("handshake finished for peer %s", remoteBzzAddress.Overlay.String())
+	if len(resp.Ack.WelcomeMessage) > 0 {
+		s.logger.Infof("peer %s greets you: $s", resp.Ack.Address.String(), resp.Ack.WelcomeMessage)
```

Comment:
For some reason, I get a strangely formatted log line.

```
INFO[2020-06-19T11:04:26+02:00] peer Underlay:"\004\177\000\000\001\006\033\237\245\003'\000%\010\002\022!\003H\372\271V\344\365v\352\tm.\363H\316T}\016\213\260\255<\216\206\252_\007\344\232\324\271'+" Signature:"\037\3318Ug\344%`V\247/\020!\355\033\003m\002\265\336\251\355\013N!\030\"\341i\217[\242\330\0003ow\336\301'\017S\232\2069hOZ\376\314\007\302\312\315\003\300B\037\025\206R\344\030xO" Overlay:"0q\256(\250\003a7\214\235\324\233\016\\g^9\276\275\031\241\202\201?\273\007f){%\024\322"  greets you: $s%!(EXTRA string=Well, Hello there!)
```

I am not sure if the complete bzz address should be printed here (with signature), maybe just overlay would be enough, as we are logging the connection details already like this:

```
INFO[2020-06-19T11:04:26+02:00] successfully connected to peer [Overlay: 3071ae28a80361378c9dd49b0e5c675e39bebd19a182813fbb0766297b2514d2, Underlay: /ip4/127.0.0.1/tcp/7071/p2p/16Uiu2HAmHZsSr34G1LgdXEu7W6xa6QJgiS9MjXHEANNayqikJVri] 
```

A problem with the pattern is that `$s` should be `%s` at the end of it.

Reply:
I think it is because the address from the ack is used, so it is printing an proto generated address which probably does not have a String() method how we would like it. It's better to use remoteBzzAddress to print the peer, and resp to print the message, maybe?

Reply:
@janos, @pradovic  - Should we just print the `remoteBzzAddress.Overlay.String()`?

Reply:
I've got the notification about the comment from @paxthemax 

> @janos, @pradovic - Should we just print the remoteBzzAddress.Overlay.String()?

But do not see it here.

My answer is yes, I think that it should be enough.